<p>What is the best way of testing a function that throws on failure? Or testing a function that is fairly immune to failure?</p>

<p>For instance; I have a I/O Completion Port class that throws in the constructor if it can't initialise the port correctly. This uses the Win32 function of CreateIoCompletionPort in the initialiser list. If the handle isn't set correctly - a non-null value - then the constructor will throw an exception. I have never seen this function fail.</p>

<p>I am pretty certain that this (and other functions like it in my code) if they fail will behave correctly, the code is 50 lines long including white-space, so my questions are</p>

<p>a) is it worth testing that it will throw<br />
b) and if it is worth testing, how to?<br />
c) should simple wrapper classes as these be unit-tested?  </p>

<p>For b) I thought about overriding CreateIoCompletionPort and passing the values through. In the unit test override it and cause it to return 0 when a certain value is passed in. However since this is used in the constructor then this needs to be static. Does this seem valid or not?</p>

<p>Thanks, G.</p>
